### PR TITLE
Feature/use new API to get insights - per method

### LINKED ISSUE #280: jetbrains - use new api for get insights - per method
related to task: https://github.com/digma-ai/digma-collector-backend/issues/525

the motivation is that the backend adds its own discovery per method, and the IDs sent by the Client might not be exact as returned from the backend - anyway - the plugin should respect the insights **per** method

### PR DESCRIPTION
**Pull request by MykolaYaremchuk**
Fixes #280

### CODE REVIEW

**Comment by MykolaYaremchuk** [comment]
@arik-dig  This PR is ready for your review.

after manual testing I figured out that:
CodeLens are not working well with DotNet project on main branch#536
CodeLens work great inside Java project and work strange with DotNet Sample.
Go to latest main branch (plugin version=2.0.61) and play with it (in some rear cases code lens are being displayed for .cs files but usually no CodeLens at all)

Bug ticket is created already for this:
https://app.zenhub.com/workspaces/digma-team-629081e5a37b4b0015ac6813/issues/gh/digma-ai/digma-intellij-plugin/536


also please double check from my PR if I have to send to the API normal "codeObjectId" without prefix ("Sample.MoneyTransfer.API.Controllers.AccountController$_$GetAccount(Int64)") or should be added "method:" at the beginning?

File: ide-common/src/main/java/org/digma/intellij/plugin/refreshInsightsTask/RefreshService.kt [hunk 1]
```diff
@@ -126,12 +125,10 @@ class RefreshService(private val project: Project) {
...
 
-    private fun checkIfDataChanged(oldInsights: List<CodeObjectInsight?>, newInsights: List<CodeObjectInsight?>): Boolean {
-        val isEqual = CollectionUtils.isEqualCollection(
-                oldInsights,
-                newInsights
-        )
-        return !isEqual
+    private fun checkIfDataChanged(oldInsights: Map<String, List<CodeObjectInsight>>, newInsights: Map<String, List<CodeObjectInsight>>): Boolean {
+        // In Kotlin, you can compare two maps for equality using the == operator.
+        // The == operator checks if the two maps have the same size, the same keys, and the same values for each key.
```
Comment (asafchen-dig) [inline ide-common/src/main/java/org/digma/intellij/plugin/refreshInsightsTask/RefreshService.kt:130]: It's a well placed comment for those of who are new to kotlin

File: ide-common/src/main/kotlin/org/digma/intellij/plugin/ui/service/ErrorsViewService.kt [hunk 2]
```diff
@@ -163,7 +164,7 @@ class ErrorsViewService(project: Project) : AbstractViewService(project) {
 
 
     private fun computeErrorsPreviewCount(documentInfoContainer: DocumentInfoContainer): Int {
-        return documentInfoContainer.allInsights.stream().filter{ it is ErrorInsight }.count().toInt()
+        return documentInfoContainer.countInsightsByType(InsightType.Errors)
```
Comment (asafchen-dig) [inline ide-common/src/main/kotlin/org/digma/intellij/plugin/ui/service/ErrorsViewService.kt:167]: Shouldn't this method access `ErrorInsight.errorCount`?

**Review by asafchen-dig — APPROVED** [review]

File: ide-common/src/main/kotlin/org/digma/intellij/plugin/ui/service/ErrorsViewService.kt (hunk 2 above)
Comment (arik-dig) [reply ide-common/src/main/kotlin/org/digma/intellij/plugin/ui/service/ErrorsViewService.kt:167]: nope, the `.count()` is on java stream after the filter been applied.

File: ide-common/src/main/java/org/digma/intellij/plugin/refreshInsightsTask/RefreshService.kt (hunk 1 above)
Comment (arik-dig) [reply ide-common/src/main/java/org/digma/intellij/plugin/refreshInsightsTask/RefreshService.kt:130]: thanks to Mykola

**Pull request merged**